src/main/java/com/example/HwLes11ANWM/services/CustomUserDetailsService.java

```diff
@@ -0,0 +1,41 @@
+package com.example.HwLes11ANWM.services;
+
+
+import com.example.HwLes11ANWM.dtos.UserDto;
+import com.example.HwLes11ANWM.models.Authority;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+
+@Service
+public class CustomUserDetailsService implements UserDetailsService {
+
+    @Autowired
+    private UserService userService;
+
+
+    @Override
+    public UserDetails loadUserByUsername(String username) {
+        UserDto userDto = userService.getUser(username);
```

Comment:
Het is niet handig om je password in je (output)dto te zetten, dus is het hier misschien handiger om de repository aan te spreken ipv de service.